EventProc/src/OverlayProducer.cxx

```diff
@@ -0,0 +1,329 @@
+		  overlayHit.setTime( overlayTime ); 
+
+		  if (! isEcalHitCollection )  //special treatment for ecal
+			outHits.push_back( overlayHit );
+		  else {
+			int overlayHitID = overlayHit.getID();
+			if ( hitMap.find( overlayHitID ) == hitMap.end() ) {
+			  hitMap[ overlayHitID ] = overlayHit;
+			  //no need to add, we just did
+			  hitMap[ overlayHitID ].updateContrib(0, overlayHit.getEdep(), overlayTime);
+			  hitMap[ overlayHitID ].setID(overlayHitID);
+			}
+			else // there's a simhit here 
+			  hitMap[ overlayHitID ].addContrib(overlayIncidentID_,overlayTrackID_,overlayPdgCode_, overlayHit.getEdep(), overlayTime);
+			// incidentID = -1000, trackID = -1000, pdgCode = 0  <-- these are set in the header for now but could be parameters
+		  }
+		  
+		} //over overlay calo simhit collection
+		  
+
+		 ldmx_log(debug) <<  "Nhits in overlay collection " << outCollNames[iColl] << ": " << outHits.size() ;
+
+		if ( isEcalHitCollection ) {
+		  for ( auto &mapHit : hitMap ) {
+            outHits.push_back( mapHit.second );
+		  }
+		}// add overlaid ecal hits as contribs rather than simhits 
+
+		
+		v_outHits.push_back(outHits);
+		
+	  } // over collections 
+
+
+	  
+	  if (! overlayFile_->nextEvent() ) {                                                                                                                                 
+		std::cerr << "Couldn't read next event!" ;                                                                                                            
+		return;
+
+
+		
+    }             //  overlayEvent->nextEvent();
+		
+	  // done. 
+	  //finally, increment book keeper of which events we've used 
+	  lastEvent_++;
+
+
+	} // over overlay events 
+	  
+
+	  		
+	// need a map: collection name, and actual hit collection
+	// actually probably need two: one for simcalo hits and omne for simtracker hits
+
+	
+	// this should be added to the sim file, so to "event"
+	for (uint iColl = 0; iColl < outCollNames.size(); iColl++) { 
+	   ldmx_log(debug) <<  "Writing " << outCollNames[iColl] << " to event bus" ;
+	  event.add(outCollNames[iColl], v_outHits[iColl]);
+	}
+	
+	return;
+  }
+  
+  void OverlayProducer::onFileOpen() {//EventFile& input) {
+	if (verbosity_ > 2) {
+	   ldmx_log(debug) <<  "[ OverlayProducer ]: onFileOpen() " ;
+	}
+	
+	//      simFile_ = std::make_unique<EventFile*> ( input );
+	//      simFile_=&input ;
+	
+	return;
+  }
+  
+  void OverlayProducer::onFileClose(EventFile&) {
+	
+	return;
+  }
+  
+  void OverlayProducer::onProcessStart() {
+	if (verbosity_ > 2) {
+	   ldmx_log(debug) <<  "[ OverlayProducer ]: onProcessStart() " ;
+	}
+
+      //based on the config parameters, set stuff up 
+      // doing this here guarantess that the overlay file(s) gets set up regardless of whether we have an input file or not; onFileOpen gets called for both modes.
+
+      rndm_  = std::make_unique<TRandom2>( seed_ );
+      rndmTime_  = std::make_unique<TRandom2>( seed_ );
```

Comment:
I know you know that you need to switch to the RNSS, but I just wanted to be complete in my review.

Reply:
So, I toyed with this a bit, and I'm not sure how to do it when I'm not using a noise generator. It looks like I need to define some Conditions object but I'm not sure how to think about this. I figure the main use case envisioned so far has been electronics noise. Perhaps @tomeichlersmith or whoever ended up implementing the random seed service can comment on how to go about this?

Reply:
*Good News*: You don't need to define a whole conditions object! Basically all RNSS does is define `seed_` for you (instead of you passing it from python). This makes it centrally generated so that we can redo an entire process with only defining one "master seed".

Since the conditions objects need to be configured themselves, the earliest (safe) place to access conditions object is in `produce`. The method that other producers have taken is leaving your random engines as `nullptr`s until they are seeded, so delete the lines referencing the rndm generators above, and then...
```
//at the beginning of OverlayProducer::produce
if (rndm_.get() == nullptr) {
  //not been seeded yet, get it from RNSS
  const auto& rnss = getCondition<RandomNumberSeedService>(RandomNumberSeedService::CONDITIONS_OBJECT_NAME);
  rndm_ = std::make_unique<TRandom2>(rnss.getSeed("OverlayProducer::rndm_"));
}
if (rndmTime_.get() == nullptr) {
  //not been seeded yet, get it from RNSS
  const auto& rnss = getCondition<RandomNumberSeedService>(RandomNumberSeedService::CONDITIONS_OBJECT_NAME);
  rndmTime_ = std::make_unique<TRandom2>(rnss.getSeed("OverlayProducer::rndmTime_"));
}
```

Reply:
Thanks! So all I was missing then, was how to refer back to the object using `OverlayProducer::rndm_` – but I would never have realized the `produce` bit myself.